freetextresponse/freetextresponse.py

```diff
@@ -464,40 +503,64 @@ def _get_submitted_message(self):
         Returns the message to display in the submission-received div
         """
         result = ''
-        if self.count_attempts > 0 and self._word_count_valid():
+        if self._word_count_valid():
             result = self.submitted_message
         return result
 
+    def _get_user_alert(self, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert(TBD) div
+        """
+        result = ''
+        if not self._word_count_valid():
+            result = self._get_word_count_message(ignore_attempts)
+        return result
+
     @XBlock.json_handler
     def submit(self, data, suffix=''):
         # pylint: disable=unused-argument
         """
         Processes the user's submission
         """
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
-            raise StandardError(
-                _(
-                    'User has already exceeded the '
-                    'maximum number of allowed attempts'
-                )
-            )
-        self.student_answer = data['student_answer']
-        if self._word_count_valid():
-            if self.max_attempts == 0:
-                self.count_attempts = 1
-            else:
-                self.count_attempts += 1
-            self._compute_score()
+        if self.max_attempts == 0 or self.count_attempts < self.max_attempts:
```

Comment:
@caesar2164 @stvstnfrd 
I'm fine with this.  My only concern is the above conditional is in three different functions.  Should we add it to all three or just submit and save events?  We could add this to save, submit and the default view functions and then pass them into any functions that require them.  Both parts of the conditional are used many times from save, submit, and default